src/batched/KokkosBatched_Axpy_Decl.hpp

```diff
@@ -0,0 +1,66 @@
+#ifndef __KOKKOSBATCHED_AXPY_DECL_HPP__
+#define __KOKKOSBATCHED_AXPY_DECL_HPP__
+
+
+/// \author Kim Liegeois (knliege@sandia.gov)
+
+#include "KokkosBatched_Util.hpp"
+#include "KokkosBatched_Vector.hpp"
+
+namespace KokkosBatched {
+
+  ///
+  /// Serial AXPY
+  ///
+  ///
+  /// y <- alpha * x + y
+  ///
+  ///
+
+  struct SerialAxpy {
+    template<typename ViewType,
+             typename alphaViewType>
+    KOKKOS_INLINE_FUNCTION
+    static int
+    invoke(const alphaViewType &alpha,
+           const ViewType &X,
+           const ViewType &Y);
+  };
+
+  ///
+  /// Team AXPY
+  ///
+
+  template<typename MemberType>
+  struct TeamAxpy {
+    template<typename ViewType,
+             typename alphaViewType>
+    KOKKOS_INLINE_FUNCTION
+    static int
+    invoke(const MemberType &member, 
+           const alphaViewType &alpha,
+           const ViewType &X,
+           const ViewType &Y);
+  };
+
+  ///
+  /// TeamVector AXPY
+  ///
+
+  template<typename MemberType>
+  struct TeamVectorAxpy {
+    template<typename ViewType,
+             typename alphaViewType>
+    KOKKOS_INLINE_FUNCTION
+    static int
+    invoke(const MemberType &member, 
+           const alphaViewType &alpha,
+           const ViewType &X,
+           const ViewType &Y);
+  };
+
+}
+
+#include "KokkosBatched_Axpy_Impl.hpp"
```

Comment:
We want to include Impl directly only if ETI is off. If ETI is on, the user includes the file in their code, we do not want them to include the *Impl file as well. They will link to Kokkos Kernels library for the Impl. This is a pattern, we seem to have missed in several files. We should go back to it.